test/true-currencies/TrueCurrencyWithPoR.test.ts

```diff
@@ -0,0 +1,184 @@
+
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT)
+    expect(await token.balanceOf(owner.address)).to.equal(balanceBefore.add(AMOUNT_TO_MINT))
+  })
+
+  it('should mint successfully when feed decimals < TrueCurrency decimals', async () => {
+    // Re-deploy a mock aggregator with fewer decimals
+    const currentTusdSupply = await token.totalSupply()
+    const validReserve = currentTusdSupply.div(exp(1, 12)).add(AMOUNT_TO_MINT)
+    const mockV3AggregatorWith6Decimals = await new MockV3Aggregator__factory(owner).deploy('6', validReserve)
+    // Set feed and heartbeat on newly-deployed aggregator
+    await token.setChainReserveFeed(mockV3AggregatorWith6Decimals.address)
+    expect(await token.chainReserveFeed()).to.equal(mockV3AggregatorWith6Decimals.address)
+    await token.setChainReserveHeartbeat(ONE_DAY_SECONDS)
+    expect(await token.chainReserveHeartbeat()).to.equal(ONE_DAY_SECONDS)
+
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT, {
+      gasLimit: 200_000,
+    })
+    expect(await token.balanceOf(owner.address)).to.equal(balanceBefore.add(AMOUNT_TO_MINT))
+  })
+
+  it('should mint successfully when feed decimals > TrueCurrency decimals', async () => {
+    // Re-deploy a mock aggregator with more decimals
+    const currentTusdSupply = await token.totalSupply()
+    const validReserve = currentTusdSupply.mul(exp(1, 2)).add(AMOUNT_TO_MINT)
+    const mockV3AggregatorWith20Decimals = await new MockV3Aggregator__factory(owner).deploy('20', validReserve)
+    // Set feed and heartbeat on newly-deployed aggregator
+    await token.setChainReserveFeed(mockV3AggregatorWith20Decimals.address)
+    expect(await token.chainReserveFeed()).to.equal(mockV3AggregatorWith20Decimals.address)
+    await token.setChainReserveHeartbeat(ONE_DAY_SECONDS)
+    expect(await token.chainReserveHeartbeat()).to.equal(ONE_DAY_SECONDS)
+
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT, {
+      gasLimit: 200_000,
+    })
+    expect(await token.balanceOf(owner.address)).to.equal(balanceBefore.add(AMOUNT_TO_MINT))
+  })
+
+  it('should mint successfully when TrueCurrency supply == proof-of-reserves', async () => {
+    // Set heartbeat to 1 day
+    await token.setChainReserveHeartbeat(ONE_DAY_SECONDS)
+    expect(await token.chainReserveHeartbeat()).to.equal(ONE_DAY_SECONDS)
+
+    // Mint TUSD
+    const balanceBefore = await token.balanceOf(owner.address)
+    await token.mint(owner.address, AMOUNT_TO_MINT)
+    expect(await token.balanceOf(owner.address)).to.equal(balanceBefore.add(AMOUNT_TO_MINT))
+  })
+
+  it('should revert if TrueCurrency supply > proof-of-reserves', async () => {
+    // Re-deploy aggregator with fewer TUSD in reserves
```

Comment:
We'r not re-deploying here, just updating the oracle's answer.

Reply:
Removed the comment.